### DISCUSSION TITLE
Fill XAxis padding

### DISCUSSION THREAD

**Discussion by nik-webdevelop** [recharts/recharts#3993] [Q&A] [answered]
What are the options to fill XAxis padding or make it look like it?

Simple Area chart with `padding={{ left: 30, right: 30 }}`

![image](https://github.com/recharts/recharts/assets/10981702/ecbfaba7-10f0-4788-a163-c1c35bc4a123)

Desired result: 

![image](https://github.com/recharts/recharts/assets/10981702/1845058c-dd0c-4c4a-99dd-80416b2f6f88)

The only option that i had come up with is to create fake items at the start/end of a data set and disable `tick` and `tooltip` for these fake items.

Is there a better approach to this?

**Comment by ckifer** [comment]
Padding is padding meaning there is nothing in that space. 

Can you explain more about what you're trying to do?

**Reply by nik-webdevelop** [reply]
@ckifer thanks for a swift response!

This is the desired result, where the chart starts not from the first data point but has some space before it

![image](https://github.com/recharts/recharts/assets/10981702/f19d3d06-cd87-44f9-b668-3670b4ec218b)

**Reply by ckifer** [reply, marked as answer]
You could try to add a null datapoint at the start and use connectNulls (although that will go to 0). 

Not sure otherwise, repeating your first and last datapoint and hiding their ticks may be your best bet. 

Since the chart continues, it looks like you have data there. You need a datapoint to start and end with

**Reply by nik-webdevelop** [reply]
Yes, I think it should work. Thank you
